addons/pos_online_payment_self_order/tests/test_self_order_frontend.py

```diff
@@ -3,10 +3,64 @@
 from unittest.mock import patch
 
 import odoo.tests
+from odoo import Command
+from odoo.addons.mail.tests.common import mail_new_test_user
+from odoo.addons.pos_online_payment.tests.online_payment_common import OnlinePaymentCommon
 from odoo.addons.pos_self_order.tests.self_order_common_test import SelfOrderCommonTest
 # from odoo.addons.pos_online_payment.models.pos_payment_method import PosPaymentMethod
 
 
 @odoo.tests.tagged("post_install", "-at_install")
 class TestSelfOrderFrontendMobile(SelfOrderCommonTest):
     pass
+
+class TestUi(OnlinePaymentCommon):
+    def _get_url(self):
+        return f"/pos/ui?config_id={self.pos_config.id}"
+
+    def start_pos_tour(self, tour_name, login="pos_admin", **kwargs):
+        self.start_tour(self._get_url(), tour_name, login=login, **kwargs)
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.payment_provider = cls.provider # The dummy_provider used by the tests of the 'payment' module.
+
+        cls.payment_provider_old_company_id = cls.payment_provider.company_id.id
+        cls.payment_provider_old_journal_id = cls.payment_provider.journal_id.id
+        cls.payment_provider.write({
+            'company_id': cls.company.id,
+        })
+        cls.online_payment_method = cls.env['pos.payment.method'].create({
+            'name': 'Online payment',
+            'is_online_payment': True,
+            'online_payment_provider_ids': [Command.set([cls.payment_provider.id])],
+        })
+
+        cls.sales_journal = cls.env['account.journal'].create({
+            'name': 'Sales Journal for POS OP Test',
+            'code': 'POPSJ',
+            'type': 'sale',
+            'company_id': cls.company.id
+        })
+        cls.pos_config = cls.env['pos.config'].create({
+            'name': 'POS OP Test Shop',
+            'module_pos_restaurant': True,
+            'invoice_journal_id': cls.sales_journal.id,
+            'journal_id': cls.sales_journal.id,
+            'payment_method_ids': [Command.link(cls.online_payment_method.id)],
+        })
+        cls.pos_admin = mail_new_test_user(
+            cls.env,
+            groups="base.group_user,point_of_sale.group_pos_manager",
+            login="pos_admin",
+            name="POS Admin",
+            tz="Europe/Brussels",
+        )
+
+
```

Comment:
Last thing and then I'm good. When running your test I had a warning message. 
> 2024-12-09 14:44:36,869 2538820 WARNING 18.0 odoo.addons.pos_online_payment_self_order.tests.test_self_order_frontend.TestSelfOrderOnlinePayment.test_01_online_payment_with_multi_table: HttpCase test should be in post_install only 
```suggestion
@odoo.tests.tagged("post_install", "-at_install")
```